components/CharmCalculator.tsx

```diff
@@ -0,0 +1,219 @@
+"use client";
+
+import { useState } from "react";
+
+// Components
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
+import { Table, TableBody, TableCaption, TableCell, TableHead, TableHeader, TableRow } from "@/components/ui/table";
+
+// Data
+import Heroes from "../public/data/heroes.json";
+
+// Helpers
+import { Charm, CharmPosition, statSummary } from "@/lib/charm";
+
+// Charm Calculator
+export default function CharmCalculator() {
+  // Properties
+  const availableClasses = Heroes.Classes.map((classType) => classType.Class);
+  const [selectedClass, setSelectedClass] = useState<String>(availableClasses[0]);
+
+  const [availableHeroes, setAvailableHeroes] = useState<any[] | null>(null);
```

Comment:
I would default this to all available heroes and refine based on the class Select - also add a `All Classes` which resets this to all heroes.